### ISSUE TITLE
check-sources.sh failing with `awk: line 4: regular expression compile failed` on mawk 1.3.3

### ISSUE THREAD

**Issue by nathanperkins** [cilium/cilium#21284]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### What happened?

`make postcheck` fails on systems with mawk 1.3.3 installed:

```
pkg/datapath/loader/check-sources.sh
awk: line 4: regular expression compile failed (missing operand)
.* |"|)|;
```

This occurs because [this line](https://github.com/cilium/cilium/blob/9f0bbbaefde6c112031906487bbc8045496ac0da/pkg/datapath/loader/check-sources.sh#L13) (introduced in [this commit](https://github.com/cilium/cilium/commit/bb5e3fef6f0ed123a5264063368692693b05757a)) doesn't properly escape one of the parens:

`{gsub(/.* |"|)|;/, "", $1); print $1}` should be `{gsub(/.* |"|\)|;/, "", $1); print $1}`

Newer versions of mawk (and other versions of awk) don't seem to have an issue with this syntax, but regex linters like regexr complain about it.

I will submit a fix for this.

### Cilium Version

Any version after https://github.com/cilium/cilium/commit/bb5e3fef6f0ed123a5264063368692693b05757a

### Kernel Version

n/a

### Kubernetes Version

n/a

### Relevant log output

```shell
+ make postcheck
pkg/datapath/loader/check-sources.sh
awk: line 4: regular expression compile failed (missing operand)
.* |"|)|;
make: *** [Makefile:625: check-sources] Error 2
+ EXIT_VALUE=2
+ clean_up
```

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by sayboras** [comment]
Thanks for reporting, feel free to shout out if you need any help on submitting fix.

**Comment by nathanperkins** [comment]
Fix is up for review: #21285
